src/overlays/actors/ovl_En_Snowman/z_en_snowman.c

```diff
@@ -42,155 +73,1123 @@ const ActorInit En_Snowman_InitVars = {
+            }
+        }
+    }
+}
+
+void EnSnowman_Update(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnSnowman* this = THIS;
+    f32 wallCheckRadius;
+
+    if (this->actionFunc != EnSnowman_SplitDoNothing) {
+        DECR(this->combineTimer);
+
+        EnSnowman_UpdateDamage(this, globalCtx);
+        this->actionFunc(this, globalCtx);
+
+        if (this->actionFunc != EnSnowman_MarkForDeath) {
+            Actor_MoveWithGravity(&this->actor);
+            if ((EN_SNOWMAN_GET_TYPE(&this->actor) == EN_SNOWMAN_TYPE_LARGE) &&
+                (this->actionFunc == EnSnowman_ReadySnowball)) {
+                wallCheckRadius = (this->skelAnime.curFrame * 0.016666668f) + 1.0f;
+                wallCheckRadius = CLAMP_MAX(wallCheckRadius, 1.3f);
+                wallCheckRadius *= this->collider.dim.radius;
+            } else {
+                wallCheckRadius = this->collider.dim.radius;
+            }
+
+            Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 30.0f, wallCheckRadius, 0.0f, 0x1D);
+            if ((this->actor.floorPoly != NULL) && ((this->actor.floorPoly->normal.y * SHT_MINV) < 0.7f)) {
+                Math_Vec3f_Copy(&this->actor.world.pos, &this->actor.prevPos);
+                if (!this->turningOnSteepSlope) {
+                    this->snowPileTargetRotY = Math_FAtan2F(this->actor.floorPoly->normal.z * SHT_MINV,
+                                                            this->actor.floorPoly->normal.x * SHT_MINV);
+                    this->turningOnSteepSlope = true;
+                }
+            } else {
+                func_800BE3D0(&this->actor, this->actor.shape.rot.y, &this->actor.shape.rot);
+            }
+
+            Collider_UpdateCylinder(&this->actor, &this->collider);
+            if (this->collider.base.acFlags & AC_ON) {
+                CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+            }
+            CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+
+            if (this->actor.draw == EnSnowman_Draw) {
+                Actor_SetFocus(&this->actor, this->actor.scale.y * 1800.0f);
+            } else {
+                Actor_SetFocus(&this->actor, this->actor.scale.y * 720.0f);
+            }
+
+            if (this->drawDmgEffAlpha > 0.0f) {
+                Math_StepToF(&this->drawDmgEffAlpha, 0.0f, 0.05f);
+                this->drawDmgEffScale = (this->drawDmgEffAlpha + 1.0f) * 0.275f;
+                this->drawDmgEffScale = CLAMP_MAX(this->drawDmgEffScale, 0.55f);
+            }
+        }
+    }
+}
+
+void EnSnowman_UpdateSnowball(Actor* thisx, GlobalContext* globalCtx) {
+    EnSnowman* this = THIS;
+    s16 scale;
+    s32 i;
+
+    if (this->timer > 0) {
+        this->timer--;
+    } else {
+        this->collider.base.ocFlags1 |= OC1_ON;
+    }
+
+    if ((this->actor.bgCheckFlags & 8) || (this->actor.bgCheckFlags & 1) || (this->actor.bgCheckFlags & 0x10) ||
```

Comment:
Hmm, interesting. I wonder if these flags are always checked like this.

Reply:
Yeah, it seems macro-y, but this is also the only instance in the entire repo of them checking the flags in this order.

Reply:
Yeah, this seems more an anomaly than anything, they check several at once elsewhere.